test/testdata/desugar/blockpass.rb

```diff
@@ -45,7 +45,6 @@ def foo(&blk)
     calls_with_object(&HasToProc.new)
     calls_with_object {|*args| HasToProc.new.to_proc.call(*args)}
     CallsWithObject.calls_with_object(&:meth)
-    CallsWithObject&.calls_with_object(&:meth)
```

Comment:
Why did we delete this line?

Reply:
Uh, good question. It was raising the new error on this line because `CallsWithObject` cannot be nil, but that raises the question of why it wasn't raising "code is unreachable" error in the first place (or maybe it was, but somehow we had marked it as expected). Will look into that.

Reply:
In any case, we should almost never delete code to make tests pass when new errors are introduced, instead of adding an additional `# error:` comment

(i.e., this implicitly was the only place in Sorbet's test suite where we were testing that Sorbet doesn't crash when you use `&.` on a constant literal receiver! so it's definitely code we want to continue to test, and it's fine if the behavior has changed.)

Reply:
(Pulling in some Slack discussion) Looks like the new desugaring for `&.` no longer triggers the bug in https://github.com/sorbet/sorbet/issues/621, which is the only reason this line worked in the first place.

I'll restore this line with a suitable `# error:`, and will watch out for any new errors when testing against Stripe code.